### PR TITLE
Redirect api

### PR DESCRIPTION
**Pull request by jaap3**
This PR adds an API endpoint to the redirects app. This is an alternative to the solution proposed in PR #6110, that doesn't mix wagtail core and contrib and works for external redirects.

### CODE REVIEW

**Comment by jaap3** [comment]
@zerolab can this be tagged headless? We've been using this to enable redirects in a headless wagtail setup, so it fits the bill ;)

**Comment by jaap3** [comment]
Just rebased to fix a merge conflict

**Comment by rohitsrma** [comment]
Hello @lb- , I would like to work on this!

**Comment by lb-** [comment]
Go for it @rohitsrma  - there has not been a comment for some time so if you are keen to pick up where this left off and review/work through feedback - that would be great.

Please reference this PR and the issue being fixed when you raise a new PR so we can acknowledge the original work done here.

**Comment by rohitsrma** [comment]
Thank you, @lb-. Could you please clarify the significance of this feature? From my current understanding, after implementing these changes, if we make a request to a path that has a redirect, the response will include a detail_url such as `'http://127.0.0.1:8000/api/v2/redirects/11/'` along with the old URL path. However, I haven't fully grasped its practical uses yet.

**Comment by jaap3** [comment]
@rohitsrma feel free to take this up. I don't have the time to work on this right now.

I've been using a version of this code in a project for a significant amount of time now (longer than this PR has been open) on a project that uses Wagtail as a headless CMS for a Vue/Nuxt frontend.

How it is supposed to work is as follows (at least in my use case):

* Frontend gets a request for a page (i.e. `/bread/brioche/`)
* Frontend uses the `/pages/find/?html_path=/bread/brioche/` endpoint to look up the page to render
    * This endpoint returns a 404, as there is no page
* Frontend uses the `/redirects/find/?html_path=/bread/brioche/` endpoint to see if there is a redirect for this url
    * This time there is a response, as there is a redirect
* Frontend uses the `location` field in the response to redirect the user to the correct location (i.e. `/pastry/brioche/`)

I think @lb- proposal basically works the same, but adds a redirect listing (which could be useful if clients want to build up a cache/pre-determine redirects) and detail url.

So in short, this feature is useful for projects that use Wagtail as a headless CMS and have frontend clients need to be able to support the redirect feature in Wagtail.

**Comment by rohitsrma** [comment]
Oh thanks @jaap3, Now i understood the whole point.

**Comment by lb-** [comment]
Closing in favor of newer https://github.com/wagtail/wagtail/pull/11510

**Comment by lb-** [comment]
Thank you @jaap3  - this has been reworked and merged in by @rohitsrma 996abeae8eaeb96677a0e6f07e6addf25ca8f698 / https://github.com/wagtail/wagtail/pull/11510
